core/src/main/scala/com/salesforce/op/dsl/RichDateFeature.scala

```diff
@@ -183,3 +185,40 @@ trait RichDateFeature {
   }
 
 }
+
+abstract class TimePeriodTransformers[T <: Date : TypeTag] {
+  val f: FeatureLike[T]
+
+  /**
+   * Converts a Date or DateTime feature into a selected time period.
+   *
+   * @param period type of [[TimePeriod]] to convert date feature to
+   * @return Integer feature of time period value
+   */
+  def toTimePeriod(period: TimePeriod): FeatureLike[Integral] = {
+    val periodFun: Long => Int = period match {
```

Comment:
Lets add a concrete transformer class that extends `UnaryTransformer` instead of using `UnaryLambdaTransformer`